core/server/src/main/java/alluxio/master/AlluxioMaster.java

```diff
@@ -241,6 +242,14 @@ protected AlluxioMaster(MasterContext masterContext) {
       Configuration.set(Constants.MASTER_RPC_PORT, Integer.toString(mPort));
       mMasterAddress = NetworkAddressUtils.getConnectAddress(ServiceType.MASTER_RPC);
 
+      // Determine port for web server
+      if (Configuration.getInt(Constants.MASTER_WEB_PORT) == 0) {
+        try (ServerSocket serverSocket = new ServerSocket(0)) {
+          int freePort = serverSocket.getLocalPort();
```

Comment:
is it possible that this port was taken after freed but before UIWebServer really launched?